src/core/client.ts

```diff
@@ -31,62 +31,61 @@ export class BTCPayClient {
     };
   }
 
-  public async pair_client(code: string): Promise<any> {
-    const re = new RegExp('^\\w{7,7}$');
+  public async pair_client(code: string): Promise<PairClientResponse> {
+    const re = new RegExp('^\\w{7}$');
 
     if (!re.test(code)) {
-      throw 'pairing code is not valid';
+      throw new Error('pairing code is not valid');
     }
 
     const payload = {
       id: this.clientId,
       pairingCode: code,
     };
 
-    try {
-      const data = await this.unsigned_request('/tokens', payload);
+    return this.unsigned_request('/tokens', payload).then((data: any) => {
       const _data = data[0];
       const _res: any = {};
       _res[_data.facade] = _data.token;
       return _res;
-    } catch (error) {
-      throw error;
-    }
+    });
   }
 
   public async get_rates(
     currencyPairs: string[],
     storeID: string,
   ): Promise<Rate[]> {
-    const _params = {
-      currencyPairs,
+    return this.signed_get_request('/rates', {
+      currencyPairs: currencyPairs.join(','),
```

Comment:
Noting some bugs: bug 2 (qs encodes arrays weird, so you must join with , like BTCPay expects.